src/test/java/org/junit/internal/runners/statements/FailOnTimeoutTest.java

```diff
@@ -34,176 +35,131 @@
  */
 @RunWith(Parameterized.class)
 public class FailOnTimeoutTest {
-    private static final long TIMEOUT = 100;
-    private static final long DURATION_THAT_EXCEEDS_TIMEOUT = 60 * 60 * 1000; //1 hour
 
-    private final TestStatement statement = new TestStatement();
-
-    private final boolean lookingForStuckThread;
-    private final FailOnTimeout failOnTimeout;
-
-    @Parameterized.Parameters(name = "lookingForStuckThread = {0}")
+    @Parameters(name = "lookingForStuckThread = {0}")
     public static Iterable<Boolean> getParameters() {
         return Arrays.asList(Boolean.TRUE, Boolean.FALSE);
     }
 
-    public FailOnTimeoutTest(Boolean lookingForStuckThread) {
-        this.lookingForStuckThread = lookingForStuckThread;
-        this.failOnTimeout = builder().withTimeout(TIMEOUT, MILLISECONDS).build(statement);
-    }
+    @Parameter
+    public boolean lookingForStuckThread;
+
+    @Test
+    public void noExceptionIsThrownWhenWrappedStatementFinishesBeforeTimeoutWithoutThrowingException()
+            throws Throwable {
+        FailOnTimeout failOnTimeout = failAfter50Ms(new FastStatement());
 
-    private FailOnTimeout.Builder builder() {
-        return FailOnTimeout.builder().withLookingForStuckThread(lookingForStuckThread);
+        failOnTimeout.evaluate();
+
+        // test is successful when no exception is thrown
     }
 
     @Test
     public void throwsTestTimedOutException() {
         assertThrows(
                 TestTimedOutException.class,
-                evaluateWithWaitDuration(DURATION_THAT_EXCEEDS_TIMEOUT));
+                run(failAfter50Ms(new RunForASecond())));
     }
 
     @Test
     public void throwExceptionWithNiceMessageOnTimeout() {
-        TestTimedOutException e = assertThrows(
-                TestTimedOutException.class,
-                evaluateWithWaitDuration(DURATION_THAT_EXCEEDS_TIMEOUT));
-        assertEquals("test timed out after 100 milliseconds", e.getMessage());
+        Exception e = assertThrows(
+                Exception.class,
```

Comment:
I thought it is irritating if this test repeats the assertion of the `throwsTestTimedOutException` and that it is also unnecessary because checking the message is independent of the type.
After looking at it again I decided to merge the two tests. I think that the exception type and its message are strongly coupled and therefore the two assertions build once conceptual assertion. Therefore it should be one test instead of two.

Reply:
SGTM. Thanks!